tensorflow/python/eager/def_function_test.py

```diff
@@ -240,6 +241,32 @@ def apply(self, x):
     m1 = MyModel()
     self.assertAllEqual(m1.apply(3.0), 6.0)
 
+  @unittest.expectedFailure
+  def testMethodAllowDynamicVariableWithoutGuards(self):
+
+    class Foo:
+
+      def __init__(self):
+        self._var = 0
+        self.trace_count = 0
+
+      def __call__(self, val):
+        self.compute(val)
+        return self._var
+
+      @def_function.function
+      def compute(self,val):
+        self.trace_count += 1
+        self._var = variables.Variable(val)
+       
+    def_function.ALLOW_DYNAMIC_VARIABLE_CREATION = True
+    foo = Foo()
+    self.assertAllEqual(foo(0.3), 0.3)
+    self.assertEqual(foo.trace_count, 2)
```

Comment:
tf.function doesn't make any guarantees on the number of times it traces the function. Best to omit this counter and its checks.

Reply:
Do you mean that we cannot claim the number of expected traces also when this case will be resolved? 
I am asking this cause with failing test we need to think a little bit forward.
If not I am ok to remove it

Reply:
P.s. just to speedup the review rounds I've removed it. But let me know what do you think about this.

Reply:
Thanks, yes, tf.function reserves the right to trace the function for any number of times in each call, and change this number without notice. For example, today it traces twice, but we'd like to reduce that number to just once.